harpoon/suite.go

```diff
@@ -310,18 +323,46 @@ func (s *Suite) RunM(m *testing.M) {
 }
 
 func (s *Suite) RunT(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	var testLog bytes.Buffer
 	if s.output != "" {
 		s.suite.Options.Output = &testLog
 		s.suite.Options.NoColors = true
 	}
 
+	var errMessage internaltesting.TerminationError
+	var wg sync.WaitGroup
+	if s.exitOnCleanupFailures {
+		s.options.ExitBehavior = internaltesting.ExitBehaviorTestFail
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			select {
+			case <-ctx.Done():
+				return
+			case errMessage = <-internaltesting.TerminationChan:
+			}
+		}()
+	}
+
 	s.suite.Options.TestingT = t
+
 	s.suite.Run()
+	cancel()
```

Comment:
If this `cancel` is invoked here why we have defer `cancel` at the top of this function?